tools/ds-identify

```diff
@@ -1601,12 +1623,39 @@ dscheck_WSL() {
     # See https://youtu.be/lwhMThePdIo?t=2431&si=JKTHx39TyRgPbzkZ and
     # https://learn.microsoft.com/en-us/windows/wsl/wsl-config#what-is-drvfs
     # for more information.
-    if [ "$(grep -c '^[^[:space:]]* [^[:space:]]* 9p [^[:space:]]*aname=drvfs;.*' "${PATH_ROOT}/proc/mounts")" = 0 ]; then
+    mountpoints=$(grep '^[^[:space:]]* [^[:space:]]* 9p [^[:space:]]*aname=drvfs;.*' "${PATH_ROOT}/proc/mounts" | cut -f2 -d' ')
+
+    if [ -z "$mountpoints" ]; then
         debug 1 "WSL datasource requires access to Windows drives mount points"
         return "${DS_NOT_FOUND}"
     fi
 
-    return "${DS_FOUND}"
+    # We know we are under WSL and have acess to the host filesystem,
+    # so let's find the .cloud-init directory
+    cloudinitdir="$(WSL_cloudinit_dir_in "$mountpoints")"
+    if [ -z "$cloudinitdir" ]; then
+        debug 1 "%USERPROFILE/.cloud-init/ directory not found"
+        return "${DS_NOT_FOUND}"
+    fi
+
+    # and the applicable userdata file. Notice the ordering in the for-loop
+    # must match our expected precedence, so the file we find is what the
+    # datasource must process.
+    instance_name="$(WSL_instance_name)"
+    distro_id=$(grep "^ID=" "${PATH_ROOT}/etc/os-release" | cut -d'=' -f2 )
```

Comment:
Typically we try to avoid parsing using external tools like grep / cut in favor of using built-in operations.

Reply:
Maybe we'd want to `source /etc/os-release` and use `{$ID}` and `${VERSION_ID}`

Reply:
>  some versions of redhat which will only contain /etc/redhat-release

@blackboxsw Which versions do you mean? I haven't seen a RHEL or derivative without it before - it is required by systemd last time I checked.

> Maybe we'd want to source /etc/os-release and use {$ID} and ${VERSION_ID}

From `man 5 os-release`:

       The format of os-release is a newline-separated list of environment-like shell-compatible
       variable assignments. It is possible to source the configuration from Bourne shell scripts,
       however, beyond mere variable assignments, no shell features are supported (this means
       variable expansion is explicitly not supported), allowing applications to read the file
       without implementing a shell compatible execution engine.

So that seems viable

Reply:
@holmanb good points. Yes that was just my recall that this was the case on some images. But specifically we added support for CentOS6 and Rawhide at the time. Neither image stream had an /etc/os-release file.   Given we don't support CentOS6 anymore because of old python and rawhide now carries /etc/os-release this is a non-issue. I agree `. /etc/os-release` instead of `source /etc/os-release` good catch on POSIX there. @CarlosNihelton I think this is approach otherwise holds water and should support any Linux distros we currently support.